Implement EOS_DISABLE for non-Windows platforms to fix build issues

JessTello opened the pull request:
Builds for iOS were failing due to EOS Plugin dependencies, even though EOS was not required on these platforms.
- Implemented conditional compilation using EOS_DISABLE for non-Windows platforms (Mac, iOS, Android)
- Fixed build failures on iOS caused by EOS Plugin when EOS_DISABLE
- Verified successful builds on Mac, Android, iOS, and Linux
- Ensured Windows integration remains unaffected

JessTello committed:
Fix: Implement EOS_DISABLE for non-Windows platforms

- Implemented conditional compilation using EOS_DISABLE for non-Windows platforms
- Fixed build failures on iOS caused by EOS Plugin when EOS_DISABLE
- Ensured Windows integration remains unaffected

matt-clarke commented on Assets/Tests/Editor/UnitTestResultsWindow.cs:
```diff
@@ -19,7 +19,7 @@
  * SOFTWARE.
  */
-
+#if !EOS_DISABLE
```
Can you add a new `.asmdef` to the folder instead, make it Editor only and add the `!EOS_DISABLE` define constraint? There should be nearby asmdefs that you can refer to for file name and config.
This then applies to all code in the assembly and avoids the need to wrap code like this

JessTello replied:
Fixed and included in the latest commit.

matt-clarke commented on com.playeveryware.eos/Runtime/Core/SystemMemory.cs:
```diff
@@ -32,6 +32,7 @@
 #define DYNAMIC_MEMORY_ALLOCATION_AVAILABLE
 #endif
 
+#if !EOS_DISABLE
```
This code is within the assembly changed below at: `Runtime/Core/com.playeveryware.eos.core.asmdef`
That now has the define constraint so there's no need to edit the code.

```suggestion
```

matt-clarke commented on com.playeveryware.eos/Runtime/Core/SystemMemory.cs:
```diff
@@ -131,4 +132,5 @@ static public extern IntPtr Mem_generic_align_realloc(IntPtr ptr, size_t size_in
-}
\ No newline at end of file
+}
+#endif
```
```suggestion
```

JessTello replied:
Fixed and included in the latest commit.

matt-clarke commented on com.playeveryware.eos/Runtime/Core/Utility/PluginVersionInterface.cs:
```diff
@@ -1,3 +1,4 @@
+#if !EOS_DISABLE
```
As per `SystemMemory.cs`
```suggestion
```

JessTello replied:
Fixed and included in the latest commit.

matt-clarke commented on com.playeveryware.eos/Runtime/Core/Utility/PluginVersionInterface.cs:
```diff
@@ -39,4 +40,5 @@ public static Utf8String GetVersion()
-}
\ No newline at end of file
+}
+#endif
```
```suggestion
```

JessTello replied:
Fixed and included in the latest commit.

matt-clarke commented on Assets/Plugins/Source/Editor/com.playeveryware.eos-Editor.asmdef:
```diff
@@ -17,7 +17,9 @@
-    "defineConstraints": [],
+    "defineConstraints": [
+        "!EOS_DISABLE"
+    ],
```
Whilst you're updating these can you also do a quick search for other `.asmdefs` in the project that need updating? I can see a number of them don't have the define constraint.

It's easy to find them by:
- Searching for `t:asmdef` in the project window in Unity
- Changing the "Search" option to "Assets"
    - I can see the ones in the package appear to be fine
    - Those can be filtered to by selecting the "Packages/Epic Online Services..." folder and then selecting that option from the "Search" dropdown

If this is complicating things by adding additional testing then don't worry for now and we can get a ticket created

JessTello replied:
Fixed and included in the latest commit.

JessTello committed:
Refactor: Update .asmdef files and add EOS_DISABLE define constraint

- Updated multiple .asmdef files.
- Added EOS_DISABLE define constraint to relevant assembly definition files.
- Modified UnitTestResultsWindow, PluginVersionInterface, and UnitTestResultsWindow to exclude EOS_DISABLE condition.
- Created new asmdef for Assets/Tests/Editor/com.playeveryware.eos.test.asmdef
- Ensured consistency across all .asmdef files using Unity's project search.

billfr-ApexSystems reviewed (approved):
Approved with requested changes
